adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/HonoBasicAuthHandler.java

```diff
@@ -56,4 +63,18 @@ protected void processException(final RoutingContext ctx, final Throwable except
 
         AuthHandlerTools.processException(ctx, exception, authenticateHeader(ctx));
     }
+
+    @Override
+    public void parseCredentials(final RoutingContext context, final Handler<AsyncResult<JsonObject>> handler) {
+        super.parseCredentials(context, ar -> {
+            if (ar.succeeded()) {
+                final SpanContext spanContext = TracingHandler.serverSpanContext(context);
+                if (spanContext != null && !(spanContext instanceof NoopSpanContext)
+                        && authProvider instanceof CredentialsApiAuthProvider) {
+                    ((CredentialsApiAuthProvider) authProvider).injectSpanContextIntoAuthInfo(spanContext, ar.result());
```

Comment:
The idea was that it only makes sense to _inject_ the span context when using an `AuthProvider` that is also able to _extract_ the span context.
Therefore both operations (inject/extract) are kept in the `CredentialsApiAuthProvider` base class.

Someone wanting to use a different `AuthProvider` (not based on `CredentialsApiAuthProvider`) would have to to the extraction manually in the `authenticate` method. I think in that case it makes more sense for that someone to use the `CredentialsApiAuthProvider` base class.

(Concerning the 2 commits below: I had already implemented a solution where the auth info JSON was always added, but I didn't like it, so I reverted that in the 2nd commit.)

Reply:
IMHO we should not restrict the possibility to pass a span context to subclasses of `CredentialsApiAuthProvider`, because it seems to be a very generic mechanism that can be employed by any `AuthProvider` implementation. FMPOV it would not be too much of a waste of CPU cycles to inject the context into the auth info even if it is never extracted by the `AuthProvider` afterwards (for the default protocol adapter configuration, this would never be the case anyway). IMHO the `TracingHelper` could simply offer two static methods for injecting/extracting a span context using a given `Tracer` and Inject/Extract adapter.

Reply:
Ok, I've moved the inject/extract methods to `TracingHelper`.